pyfaaster/aws/dynamodb.py

```
@@ -2,20 +2,48 @@
 # Copyright (c) 2016-present, CloudZero, Inc. All rights reserved.
 # Licensed under the BSD-style license. See LICENSE file in the project root for full license information.
 
+import re
+from boto3.dynamodb.types import TypeSerializer, TypeDeserializer
 
-def update_item_from_dict(table, key, dictionary):
-    """Update the item identified by `key` in the DynamoDB `table` by adding
+pattern = re.compile(r'[\W_]+', re.UNICODE)
+
+
+def update_item_from_dict(table_name, key, dictionary, client):
+    """
+    Update the item identified by `key` in the DynamoDB `table` by adding
     all of the attributes in the `dictionary`.
+    Args:
+        table_name (str):
+        key (dict):
+        dictionary (dict):
+        client:
+
+    Returns:
+        dict
     """
+    serializer = TypeSerializer()
+    deserializer = TypeDeserializer()
+
+    # Convert all keys to alphanumeric
+    dictionary = {pattern.sub("", k): v for k, v in dictionary.items()}
```

This function should not change the dictionary keys to only alphanumeric; it should _only_ mutate the values _into_ the attribute expression names and values. If a dictionary key/value really is unsuitable for dynamo, e.g. a dictionary key is 256 characters long, this method should throw an exception.